kernel/src/main/java/org/kframework/parser/json/JsonParser.java

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2018 K Team. All Rights Reserved.
+package org.kframework.parser.json;
+
+import org.kframework.kore.K;
+import static org.kframework.kore.KORE.KLabel;
+import org.kframework.kore.mini.InjectedKLabel;
+import org.kframework.kore.mini.KApply;
+import org.kframework.kore.mini.KRewrite;
+import org.kframework.kore.mini.KSequence;
+import org.kframework.kore.mini.KToken;
+import org.kframework.kore.mini.KVariable;
+import org.kframework.parser.outer.Outer;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.io.StringReader;
+import java.util.Arrays;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+/**
+ * Parses a Json term into the KORE data structures.
+ */
+public class JsonParser {
+
+    public static final String KTOKEN         = "KToken"
+                             , KAPPLY         = "KApply"
+                             , KSEQUENCE      = "KSequence"
+                             , KVARIABLE      = "KVariable"
+                             , KREWRITE       = "KRewrite"
+                             , INJECTEDKLABEL = "InjectedKLabel"
+                             ;
+
+    public static K parse(byte[] data) throws UnsupportedEncodingException {
```

Comment:
This should really catch UnsupportedEncodingException and rethrow it as an AssertionError because otherwise it pollutes the type of the function with a throws clause even though UTF-8 is a required encoding on all conforming implementations of the JVM.